core/server/master/src/main/java/alluxio/master/file/DefaultBlockDeletionContext.java

```diff
@@ -19,20 +19,21 @@
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * Class for tracking which blocks need to be deleted, and deleting them on close.
  */
 public final class DefaultBlockDeletionContext implements BlockDeletionContext {
   private final List<BlockDeletionListener> mListeners;
-  private final List<Long> mBlocks;
+  private final ConcurrentHashMap.KeySetView<Long, Boolean> mBlocks;
```

Comment:
Can we just use a concurrent list or queue? I think `ConcurrentLinkedQueue` might work?

Reply:
I don't think no matter how much we try to write this unit test, it's not really going to fail on the github actions machines.

Reply:
I can do ConcurrentLinkedQueue. Usually though the ConcurrentHashMap related implementations tend to be the cleanest.